### PR TITLE
feat(overflow-menu): sync with @carbon/react v11

### PR DESCRIPTION
**Pull request by ariellalgilmore**
### Related Ticket(s)

#10078 #10117 
### Description

Sync with overflow-menu react

### Changelog

**Changed**

- updated storybook container to closely align to the same classes as react
- updated overflow-menu to extend icon-button
- remove alignment option in exchange for flipped
- removed direction options left/right

### CODE REVIEW

**Review by emyarod — COMMENTED** [review]
the menu alignment seems to always be horizontally centered and covering the button?

https://user-images.githubusercontent.com/8265238/234958926-027e62b1-5d64-44f2-a76b-a27e0bb247b5.mov

**Comment by ariellalgilmore** [comment]
@emyarod oops fixed!

**Review by emyarod — COMMENTED** [review]
it looks like the menu doesn't open with <kbd>Enter</kbd> unlike the upstream component behavior (the menu opening successfully at the end of the video is triggered by <kbd>Space</kbd>)

https://user-images.githubusercontent.com/8265238/234977823-97b808c1-07b8-49cc-ae7e-db1581eb0871.mov

**Comment by ariellalgilmore** [comment]
@emyarod ahh good catch! updated again lol!

File: packages/carbon-web-components/src/components/overflow-menu/overflow-menu.scss [hunk 1]
```diff
@@ -84,10 +84,22 @@ $css--plex: true !default;
   outline: none;
 }
 
+:host(#{$prefix}-overflow-menu-item[size='sm']) {
+  height: rem(32px);
```
Comment (kennylam) [inline packages/carbon-web-components/src/components/overflow-menu/overflow-menu.scss:88]:
```suggestion
  height: $spacing-07;
```

File: packages/carbon-web-components/src/components/overflow-menu/overflow-menu.scss [hunk 2]
```diff
@@ -84,10 +84,22 @@ $css--plex: true !default;
...
+}
+
+:host(#{$prefix}-overflow-menu-item[size='lg']) {
+  height: rem(48px);
```
Comment (kennylam) [inline packages/carbon-web-components/src/components/overflow-menu/overflow-menu.scss:92]:
```suggestion
  height: $spacing-09;
```

File: packages/carbon-web-components/src/components/overflow-menu/overflow-menu.ts
Comment (kennylam) [inline packages/carbon-web-components/src/components/overflow-menu/overflow-menu.ts:1]: A few prefixes in here that need to be changed to `CDS`.

File: packages/carbon-web-components/src/components/text-input/defs.ts [hunk 3]
```diff
@@ -35,6 +35,45 @@ export enum INPUT_SIZE {
   EXTRA_LARGE = 'xl',
 }
 
+export enum INPUT_TOOLTIP_ALIGNMENT {
```
Comment (kennylam) [inline packages/carbon-web-components/src/components/text-input/defs.ts:38]: Comments should describe alignment.

File: packages/carbon-web-components/src/components/floating-menu/floating-menu.ts
[the remark this answers is not in the document]
Comment (ariellalgilmore) [reply packages/carbon-web-components/src/components/floating-menu/floating-menu.ts:1]: gah sorry! resolved!

**Review by emyarod — COMMENTED** [review]
when the trigger button is focused the tooltip doesn't appear
it also looks like the icon is off center? adding a screenshot of the upstream component for comparison

image

upstream:
image

**Comment by ariellalgilmore** [comment]
@emyarod thankkk you! updated again!

**Review by RichKummer — APPROVED** [review]
LGTM!

**Pull request merged**